### PR TITLE
Rewrite existing post deployment tests in Playwright

### PR DESCRIPTION
**Pull request by LAKSHMIRPILLAI**
## What are you doing in this PR?

 Convert existing post deployment tests over to new framework for 2-step checkout which is the current default

[**Trello Card**](https://trello.com/c/LDi8W6o3/1622-put-two-step-only-to-100-convert-tests-over-to-new-framework-for-2-step)

## Why are you doing this?

This is part of making the 2-step only checkout the default/ changing it to 100%
We will need to convert all tests  currently in Selenium to new Playwright framework 
This includes 9 of the tests (Excluding the ones for Signed-in users) like

1)Basic page loading tests for -Newspaper Subscriptions, Guardian Weekly, Guardian Weekly Gift and Subscriptions
2)Recurring Contribution tests for 
Monthly contribution sign-up with Stripe - GBP
Monthly contribution sign-up with direct debit - GBP
Annual contribution sign-up with Stripe - USD
3)One -Off Contribution tests for 
One-off contribution sign-up with Stripe - AUD
Check browser navigation to paypal

### Things to note:
This PR also modifies the url used in contribution tests to https://support.theguardian.com/uk/contribute (Previously it was https://support.theguardian.com/uk/contribute#ab-twoStepCheckoutWithNudgeBelow=variant_a)

Also the PR includes the changes made to the wording from One-Off Contributions -which is One-time(instead of Single)

Sometimes,  some tests fail while running tests all together parallel / locally with the issue "We were rate-limited from checking if your requested Payment Request options are allowed. Please test again before going live. in " 
This passes later on retries/running individually

image

## Is this an AB test?
- [ ] Yes
- [x ] No

## Screenshots
image

### CODE REVIEW

File: support-e2e/tests/recurringContributions.test.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    await page.getByLabel("State").type("NY");
+    await page.getByLabel("ZIP code").type("90210");
+
+    await page.getByRole('radio', { name: 'Credit/Debit card' }).click();
+
+    await fillInCardDetails(page);
+
+    await checkRecaptcha(page);
+
+    await page.locator('button:has-text("Support us with")').click();
```
Comment (jamesgorrie) [inline support-e2e/tests/recurringContributions.test.ts:81]: We could use something similar to the [`Pay` button](https://github.com/guardian/support-frontend/pull/5378/files#diff-bde754ee4b521f530247ee58d848229a88f71277919bcd657ed2d0bfc001c03fR54):
```ts
    await page.getByText(/Support us with £([0-9]+) per year/).click();
```

**Review by jamesgorrie — APPROVED** [review]
The main comments are on how we use locators, happy to take this out as a discussion and maybe have some follow up PRs.

Great work on giving us a good set of coverage!

File: support-e2e/tests/oneOffContributions.test.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  const baseUrlWithFallback = baseURL ?? "https://support.theguardian.com";
+  // We should remove the forcing into the ab test once this has been made live
+  const pageUrl = `${baseUrlWithFallback}/au/contribute`;//to be updated after merging the changes to making the two step as default
+
+  const domain = new URL(pageUrl).hostname;
+  await setTestCookies(context, firstName, domain);
+  await page.goto(pageUrl, { waitUntil: "networkidle" });
+});
+
+test.describe("Sign up for a one-off contribution (New Contributions Flow)", () => {
```
[the remark this answers is not in the document]
Comment (LAKSHMIRPILLAI) [reply support-e2e/tests/oneOffContributions.test.ts:17]: This is something I copied from the old selenium tests-I will modify that for two step checkout

File: support-e2e/tests/oneOffContributions.test.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  test("One-off contribution sign-up with Stripe - AUD", async ({ page}) => {
+
+    await page.getByRole("tab").getByText("One-time").click();
+
+    await page.locator('label[for=\'amount-other\']').click();
```
[the remark this answers is not in the document]
Comment (LAKSHMIRPILLAI) [reply support-e2e/tests/oneOffContributions.test.ts:22]: I had tried to use  await page.getByRole("radio", { name: "Choose your amount" }).click(); for this , but it was not working , as it was a radio  with the name :amount and there were a number of similar  buttons with same radio type and name. So thought would be better to go with the locator 
image

File: support-e2e/tests/oneOffContributions.test.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    await page.getByLabel("Email address").type(email);
+
+    await page.getByRole('radio', { name: 'Credit/Debit card' }).click();
+
+    await fillInCardDetails(page);
+
+    await checkRecaptcha(page);
+
+    await page.locator('button:has-text("Support us with")').click();
```
[the remark this answers is not in the document]
Comment (LAKSHMIRPILLAI) [reply support-e2e/tests/oneOffContributions.test.ts:35]: Sorry, I did not quite catch this.Could you explain a bit? The test for One-Off is written for AUS- using Other amount field with an amount of 22.5 AUD

File: support-e2e/tests/oneOffContributions.test.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    await page.getByRole("button", { name: "Continue to checkout" }).click();
+
+    await page.getByLabel("Email address").type(email);
+
+    await page.getByRole('radio', { name: 'Paypal' }).click();
+
+    await page.locator('button:has-text("Pay")').click();
+
+    await expect(page).toHaveURL(/.*paypal.com/,{ timeout: 600000 });
+
```
[the remark this answers is not in the document]
Comment (LAKSHMIRPILLAI) [reply support-e2e/tests/oneOffContributions.test.ts:54]: Yes, it is difficult to test third party service.So I did it exactly as we have done the test For PayPal in old  selenium test

File: support-e2e/tests/recurringContributions.test.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    await checkRecaptcha(page);
+
+    await page.getByText(/Pay £([0-9]+) per month/).click();
+
+    await expect(page).toHaveURL(/\/uk\/thankyou/, { timeout: 600000 });
+  });
+});
+
+test.describe("Sign up for a Recurring Contribution (New Contributions Flow)", () => {
+  test("Monthly contribution sign-up with direct debit - GBP", async ({ page }) => {
```
[the remark this answers is not in the document]
Comment (LAKSHMIRPILLAI) [reply support-e2e/tests/recurringContributions.test.ts:42]: yes, thank you, updated

File: support-e2e/tests/recurringContributions.test.ts (hunk 1 above)
Comment (LAKSHMIRPILLAI) [reply support-e2e/tests/recurringContributions.test.ts:81]: Actually I had tried this, but somehow it  was not working.Tried multiple times, but it failed with error 
image

**Pull request merged**